### PR TITLE
Improve docs for `mark`/`commit`/`revert`

### PR DESCRIPTION
**Pull request by jeremiah-corrado**
Improves the docs for the `mark`/`commit`/`revert` methods on fileReader/fileWriter:

- Minor wording/clarity improvements
- The meaning of `revert` and `commit` were swapped. This is corrected
- Some method links are shortened with `~` to make docs more readable
- Clarify that a fileReader/fileWriter must be manually locked before calling `mark` even if it has `locking=true`

Per the discussion [here](https://github.com/chapel-lang/chapel/issues/19611), also adds a comment about our desire to support mark/commit/revert via a context manager in the future.

- [x] inspected built docs

### CODE REVIEW

File: modules/standard/IO.chpl [hunk 1]
```diff
@@ -4353,15 +4353,14 @@ private inline proc markHelper(fileRW) throws {
...
-   The *mark stack* stores several fileReader offsets. For any fileReader offset
-   that is between the minimum and maximum value in the *mark stack*, I/O
-   operations on the fileReader will keep that region of the file buffered in
-   memory so that those operations can be undone. As a result, it is possible to
-   perform *I/O transactions* on a fileReader. The basic steps for an *I/O
-   transaction* are:
+   The *mark stack* stores several file offsets. The ``fileReader`` will keep
+   the region of the file between its minimum and maximum *mark stack* values
+   buffered in memory so that IO operations can be undone. As a result, it is
+   possible to perform *I/O transactions* on a fileReader. The basic steps for
```
[the remark this answers is not in the document]
Comment (lydia-duncan) [reply modules/standard/IO.chpl:4362]: Alternatively, we could link to the type here and/or on line 4359.

File: modules/standard/IO.chpl [hunk 2]
```diff
@@ -4386,35 +4385,38 @@ private inline proc markHelper(fileRW) throws {
     memory space requirements.
 
   :returns: The offset that was marked
-  :throws SystemError: if marking the fileReader failed
+  :throws SystemError: if marking the ``fileReader`` failed
  */
 proc fileReader.mark() throws do return markHelper(this);
+// In the future, we'd like to support IO transactions using context managers.
```
Comment (lydia-duncan) [inline modules/standard/IO.chpl:4391]: Let's slap a TODO on this with a date, just to make it easier to search for and stand out more visually

File: modules/standard/IO.chpl [hunk 3]
```diff
@@ -4386,35 +4385,38 @@ private inline proc markHelper(fileRW) throws {
...
+// Under this proposal, entering a transaction context would mark the channel,
+// leaving the context normally would commit, and leaving via throwing an error
+// would revert.
+// See: #19611
 
 /*
-   *Mark* a fileWriter - that is, save the current offset of the fileWriter
-   on its *mark stack*.
+   *Mark* a ``fileWriter`` - that is, save the current offset of the
+   ``fileWriter`` on its *mark stack*.
```
Comment (lydia-duncan) [inline modules/standard/IO.chpl:4399]: Similarly, this could also be a link to the fileWriter type

File: modules/standard/IO.chpl [hunk 4]
```diff
@@ -4386,35 +4385,38 @@ private inline proc markHelper(fileRW) throws {
...
       calling :proc:`fileWriter.revert`. Subsequent I/O operations will work
       as though nothing happened.
 
-   If a fileWriter has ``locking==true``, :proc:`fileWriter.mark` should be
-   called only once the fileWriter has been locked with
-   :proc:`fileWriter.lock`.  The fileWriter should not be unlocked with
-   :proc:`fileWriter.unlock` until after the mark has been committed with
-   :proc:`fileWriter.commit` or reverted with :proc:`fileWriter.revert`.
+   If a fileWriter has ``locking==true``, :proc:`~fileWriter.mark` should only
+   be called only once it has been locked with :proc:`~fileWriter.lock`. The
```
Comment (lydia-duncan) [inline modules/standard/IO.chpl:4416]: Ditto comments on the fileReader documentation

File: modules/standard/IO.chpl [hunk 5]
```diff
@@ -4424,57 +4426,66 @@ proc fileReader.mark() throws do return markHelper(this);
     memory space requirements.
 
   :returns: The offset that was marked
-  :throws SystemError: if marking the fileWriter failed
+  :throws SystemError: if marking the ``fileWriter`` failed
  */
 proc fileWriter.mark() throws do return markHelper(this);
+// In the future, we'd like to support IO transactions using context managers.
```
Comment (lydia-duncan) [inline modules/standard/IO.chpl:4432]: Similarly, let's also slap a TODO and a date on this

File: modules/standard/IO.chpl [hunk 6]
```diff
@@ -4424,57 +4426,66 @@ proc fileReader.mark() throws do return markHelper(this);
...
 inline proc fileWriter.revert() {
   qio_channel_revert_unlocked(_channel_internal);
 }
 
 /*
-   Commit an *I/O transaction*. See :proc:`fileReader.mark`.  This
-   function will pop the last element from the *mark stack* and then
-   set the fileReader offset to the popped offset.
+   Commit an *I/O transaction* by popping from the ``fileReader``'s *mark stack*
+   and leaving its position unchanged. See :proc:`fileReader.mark` for a full
```
Comment (lydia-duncan) [inline modules/standard/IO.chpl:4468]: Maybe this should say "and leaving its position in the file unchanged", to distinguish between the position in the file and the position in the mark stack?  Up to you, though

**Review by lydia-duncan — APPROVED** [review]
Thanks for taking this on, just some minor suggestions that you can take or leave

File: modules/standard/IO.chpl (hunk 2 above)
Comment (jeremiah-corrado) [reply modules/standard/IO.chpl:4391]: Today's date, or the date that we would expect to start working on this?

File: modules/standard/IO.chpl [hunk 7]
```diff
@@ -4372,11 +4371,11 @@ private inline proc markHelper(fileRW) throws {
       calling :proc:`fileReader.revert`. Subsequent I/O operations will work
       as though nothing happened.
 
-   If a fileReader has ``locking==true``, :proc:`fileReader.mark` should be
-   called only once the fileReader has been locked with
-   :proc:`fileReader.lock`.  The fileReader should not be unlocked with
-   :proc:`fileReader.unlock` until after the mark has been committed with
-   :proc:`fileReader.commit` or reverted with :proc:`fileReader.revert`.
+   If a fileReader has ``locking==true``, :proc:`~fileReader.mark` should only
+   be called only once it has been locked with :proc:`~fileReader.lock`. The
```
[the remark this answers is not in the document]
Comment (jeremiah-corrado) [reply modules/standard/IO.chpl:4375]: Good point, I'll remove the `~` on `fileReader.lock` to get rid of the ambiguity.

File: modules/standard/IO.chpl (hunk 2 above)
Comment (lydia-duncan) [reply modules/standard/IO.chpl]: Today's date

**Review by lydia-duncan — APPROVED** [review]
Thanks!

**Pull request merged**